options.go

```diff
@@ -287,3 +292,18 @@ func parseVsocks(devices []string) ([]firecracker.VsockDevice, error) {
 func createFifoFileLogs(fifoPath string) (*os.File, error) {
 	return os.OpenFile(fifoPath, os.O_CREATE|os.O_APPEND|os.O_WRONLY, 0644)
 }
+
+func getSocketPath(path string) (string, error) {
+	// if user supplies path, use it
+	if path != "" {
+		return path, nil
+	}
+	// otherwise use a tempdir
+	file, err := ioutil.TempFile("", "firecracker-socket")
+	if err != nil {
+		return "", err
+	}
+	file.Close()
+	os.Remove(file.Name())
```

Comment:
I'm really confused...why are we removing the file?

Reply:
As discussed moved to a different impl that doesn't need this.